src/Sarif.WorkItemFiling/WorkItemFiler.cs

```diff
@@ -9,6 +9,8 @@
 using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Sarif.WorkItemFiling.Filtering;
```

Comment:
Not sure I buy into these namespaces as opposed to a single namespace. This approach is very appealing when authoring the code, as it keeps things conceptually distinct. API consumers are less enthusiastic, as they are required to add multiple namespaces to their source files (already cluttered with namespaces) to make use of happy path scenarios.

Reply:
See my uber-response to your namespace comments in AzureDevOpsFilingTarget.cs. Let's discuss when you get back.

---
In reply to: [306115401](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306115401) [](ancestors = 306115401)

Reply:
I don't want an overly granular namespacing strategy for the API. Yes, we can talk when I get back. I'm glad to make this change myself if you don't want to do it. My strategy is to keep things flat and to make namespaces argue their way way.

---
In reply to: [306311176](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306311176) [](ancestors = 306311176,306115401)

Reply:
See my response in AzureDevopsFilingTarget.cs I have removed the nested namespaces but preserved the directory structure.

---
In reply to: [306564912](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306564912) [](ancestors = 306564912,306311176,306115401)